CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a straight.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a straight, false otherwise.</returns>
+        private static bool IsStraight(List<Card> cards)
+        {
+            // Special case: 2345A
+            if (cards[0].GetPokerValue() == 2 && cards[1].GetPokerValue() == 3 && cards[2].GetPokerValue() == 4 && cards[3].GetPokerValue() == 5 && cards[4].IsAce())
+            {
+                return true;
+            }
+
+            for (int i = 1; i < cards.Count; i++)
+            {
+                if (cards[i].GetPokerValue() != cards[i - 1].GetPokerValue() + 1)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a three of a kind.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a three of a kind, false otherwise.</returns>
+        private static bool IsThreeOfAKind(List<Card> cards)
+        {
+            return cards[2].SameVal(cards[0]) || cards[2].SameVal(cards[4]);
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a two pair.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a two pair, false otherwise.</returns>
+        private static bool IsTwoPair(List<Card> cards)
+        {
+            int pairs = 0;
+            for (int i = 1; i < cards.Count; i++)
+            {
+                if (cards[i].SameVal(cards[i - 1]))
+                {
+                    pairs++;
+                }
+            }
+            return pairs == 2;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a pair.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a pair, false otherwise.</returns>
+        private static bool IsPair(List<Card> cards)
+        {
+            for (int i = 1; i < cards.Count; i++)
+            {
+                if (cards[i].SameVal(cards[i - 1]))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Call each of the functions used to determine if our hand is of a certain rank back to back. 
+        /// Requires that cards is of length 5, and has had SortedHand called on it.
+        /// </summary>
+        /// <param name="cards">The hand we are trying to determine the rank of.</param>
+        /// <returns>the cards rank.</returns>
+        private static Ranking GetRanking(List<Card> cards)
```

Comment:
Let's move this function above all of the "Is" methods. It will be a little cleaner.

Not like a big deal, just easier to read.